tests/test_gates.py

```diff
@@ -19,39 +79,25 @@
 )
 def test_bell_states(init_state, final_state):
     state = prepare_state(len(init_state), init_state)
-    state = apply_gate(state, H(target_idx=0))
-    state = apply_gate(state, NOT(target_idx=1, control_idx=0))
+    state = apply_gate(state, H(target=0))
+    state = apply_gate(state, NOT(target=1, control=0))
     assert jnp.allclose(state.flatten(), final_state)
```

Comment:
Maybe add an `equivalent_state` function computing the overlap as done in Qadence and I think also PyQ? Also with a default `atol` closer to `~1E-14`.

I see that there is one defined in the utils but with a different logic only for product states.

Reply:
added